docs/cudf/source/internals.md

```diff
@@ -1,29 +1,132 @@
 cuDF internals
 ==============
 
-ColumnAccessor
---------------
+The cuDF API closely matches that of the [Pandas](https://pandas.pydata.org/) library.
+Thus, we have the types `cudf.Series`, `cudf.DataFrame` and `cudf.Index` which look and
+feel very much like their Pandas counterparts.
+
+Under the hood, however, cuDF uses data structures very different from Pandas. In this document,
+we describe these internal data structures.
+
+## Column
+
+Columns are cuDF's core data structure and they are modeled after
+the [Apache Arrow Columnar Format](https://arrow.apache.org/docs/format/Columnar.html).
+
+A column represents a sequence of values, any number of which may be "null". Columns are
+specialized based on the type of data they contain. Thus we have `NumericalColumn`, `StringColumn`,
+`DatetimeColumn`, etc.,
+
+A column is composed of the following:
+
+* A **data type**, specifying the type of each element.
+* A **data buffer** that may store the data for the column elements.
+  Some column types do not have a data buffer, instead storing data in the children columns.
+* A **mask buffer** whose bits represent the validity (null or not null) of each element.
+  Columns whose elements are all "valid" may not have a mask buffer. Mask buffers are padded
+  to 64 bytes.
+* A tuple of **children** columns, which enable the representation complex types such as
+  columns with non-fixed width elements (e.g., strings, lists).
+* A *size* indicating the number of elements in the Column.
+* An integer **offset**: a column may represent a "slice" of another column,
+  in which case this offset represents the first element of the slice. The size of
+  the column then gives the extent of the slice. A column that is not a slice
+  has an offset of 0.
+
+For example, the `NumericalColumn` backing a Series with 1000 elements of type 'int32'
+and containing nulls is composed of:
+
+1. A data buffer of size 4000 bytes
+2. A mask buffer of size 128 bytes
```

Comment:
Do we want to add why these are the exact sizes (or maybe only the mask size)?
Something like
```
1. A data buffer of size 4000 bytes (sizeof(int32) * 1000)
2. A mask buffer of size 128 bytes (1000/8 padded to a multiple of 64 bytes)
```

Reply:
Great suggestion - changed it to just that